src/content/docs/analyzer/plugins.mdx

```diff
@@ -0,0 +1,52 @@
+---
+title: Analyzer Plugins
+description: An overview of how to use Biome's Analyzer Plugins
+---
+
+:::note
+Plugins are currently only available in the [2.0 beta release](/blog/biome-v2-0-beta).
+:::
+
+Biome Analyzer supports [GritQL](/reference/gritql/) plugins. Currently, these
+plugins allow you to match specific code patterns and register customized
+diagnostic messages for them.
+
+An example of a plugin is:
+
+```grit
+`$fn($args)` where {
+    $fn <: `Object.assign`,
+    register_diagnostic(
+        span = $fn,
+        message = "Prefer object spread instead of `Object.assign()`"
+    )
+}
+```
+
+You can put a GritQL snippet in a file anywhere in your project, but be mindful
+you use the `.grit` extension. Then, you can simply enable it as a plugin with
+the following configuration:
+
+```
+{
+    "plugins": ["./path-to-plugin.grit"]
+}
+```
```

Comment:
``````suggestion
```json name="biome.json"
{
    "plugins": ["./path-to-plugin.grit"]
}
```
``````

Also, users might want to publish their plugins as npm packages, so we should provide instructions on how they do so and how they can import them. If they can't, we should say it somewhere.

Reply:
Yeah, good point. I haven't really tested if it works with NPM packages yet. Maybe something to clear up before the final 2.0 release.